src/main/java/uk/gov/register/filters/DeprecatedFormatFilter.java

```diff
@@ -0,0 +1,101 @@
+        this.httpServletResponse = httpServletResponse;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+        final UriInfo uriInfo = requestContext.getUriInfo();
+        final URI requestedUri = uriInfo.getRequestUri();
+        final MultivaluedMap<String, Object> headers = responseContext.getHeaders();
+
+        extractMediaType(headers).ifPresent(mediaType -> filterMediaType(mediaType, requestedUri, headers));
+    }
+
+    private void filterMediaType(MediaType mediaType, URI requestedUri, MultivaluedMap<String, Object> headers) {
+        final String alternateUri;
+        final String alternateType;
+
+        if (recommendJson(mediaType)) {
+            alternateUri = alternateLink(requestedUri, ".json");
+            alternateType = MediaType.APPLICATION_JSON;
+        } else if (recommendCsv(mediaType)) {
+            alternateUri = alternateLink(requestedUri, ".csv");
+            alternateType = ExtraMediaType.TEXT_CSV;
+        } else {
+            return;
+        }
+
+        HttpServletResponseAdapter httpServletResponseAdapter = new HttpServletResponseAdapter(httpServletResponse);
+
+        Map<String, String> extra = new HashMap<>();
+        extra.put("rel", "alternate");
+        extra.put("type", alternateType);
+        httpServletResponseAdapter.addLinkHeader(extra, alternateUri);
+
+        httpServletResponseAdapter.setHeader(
+                HttpHeaders.WARNING,
+                warningHeader(mediaType.getSubtype())
+        );
+    }
+
+    private String alternateLink(URI requestedUri, String recommendedExtension) {
+        final String path = requestedUri.getPath();
+        final UriBuilder builder = UriBuilder.fromUri(requestedUri);
+
+        // Even if an extension was specified in the original URI, it has been stripped out
+        // at this point, so just add the new one (before any trailing slash).
+        final String newPath = path.replaceFirst("(/)?$", recommendedExtension + "$1");
+
+        builder.replacePath(newPath);
+
+        return builder.toTemplate();
+    }
+
+    private String warningHeader(String deprecatedType) {
+        return "299 - \"" + deprecatedType + " is deprecated and will be removed. See 'Link' header for a format to use instead.\"";
+    }
+
+    private boolean recommendJson(MediaType mediaType) {
+        return mediaType.equals(ExtraMediaType.TEXT_TTL_TYPE) ||
+                mediaType.equals(ExtraMediaType.TEXT_YAML_TYPE);
+    }
+
+    private boolean recommendCsv(MediaType mediaType) {
+        return mediaType.equals(ExtraMediaType.TEXT_TSV_TYPE) ||
+                mediaType.equals(ExtraMediaType.APPLICATION_SPREADSHEET_TYPE);
+    }
+
+    private Optional<MediaType> extractMediaType(MultivaluedMap<String, Object> headers) {
+        try {
+            MediaType mediaType = (MediaType) headers.getFirst(HttpHeaders.CONTENT_TYPE);
```

Comment:
I'm not sure about this actually... it feels safer to use the MediaType object for the comparison as I'm not sure if the string representation of the value is always going to be exactly what I expected. If I use MediaType.equals it should do the right thing even if there's differences in case or whitespace.

Let me know if there's a better way to accomplish this though.